flowmachine/flowmachine/features/subscriber/active_subscribers.py

```diff
@@ -0,0 +1,181 @@
+        If provided, string or list of string which are msisdn or imeis to limit
+        results to; or, a query or table which has a column with a name matching
+        subscriber_identifier (typically, msisdn), to limit results to.
+    minor_period_count: int, default 24
+        The number of minor periods to split the major period into
+    minor_period_length: int, default 1
+        The number of period_units that make up a minor period
+    period_unit: {'days','hours','minutes'} default 'hours'
+        The unit of time to of minor_period_length
+
+    Notes
+    -----
+    * The date range will be inclusive os `start_date` but exclusive of `end_date` (ie range = start_
+    date ... (end_date - 1 second))
+    * The default values for minor_period_count and minor_period_length assume you wish to seach
+    for subscribers who are active at least `minor_period_threshold` hours throughout the day.
+
+    Examples
+    --------
+    Returns subscribers who were active on at least three hours between 2016-01-01 and 2016-01-02
+
+
+    >>>     active_subscribers = ActiveSubscribers(
+                start_date=date(year=2016, month=1, day=1),
+                end_date=date(year=2016, month=1, day=2),
+                minor_period_threshold=3,
+                major_period_threshold=1,
+                tables=["events.calls"],
+            )
+
+    Returns subscribers who were active in at least five hours, at least three days out of the four
+    between 2016-01-01 and 2016-01-04
+
+
+    >>>     active_subscribers = ActiveSubscribers(
+                start_date=date(year=2016, month=1, day=1),
+                end_date=date(year=2016, month=1, day=4),
+                minor_period_threshold=5,
+                major_period_threshold=3,
+                tables=["events.calls"],
+            )
+
+    Returns subscribers that were active in at least two ten minute intervals within half an hour,
+    at least three times across the two hours between 20:00:00 and 22:00:00 on 2016-01-01
+
+
+    >>> active_subscribers = ActiveSubscribers(
+                start_date=datetime(year=2016, month=1, day=1, hour=20),
+                end_date=datetime(year=2016, month=1, day=1, hour=22),
+                minor_period_threshold=2,
+                major_period_threshold=3,
+                tables=["events.calls"],
+                minor_period_count=3,
+                minor_period_length=10,
+                period_unit="minutes",
+            )
+
+
+    """
+
+    period_to_rrule_mapping = {
+        "days": rr.DAILY,
+        "hours": rr.HOURLY,
+        "minutes": rr.MINUTELY,
+    }
+
+    def __init__(
+        self,
+        start_date: Union[date, datetime, str],
+        end_date: Union[date, datetime, str],
+        minor_period_threshold: int,
+        major_period_threshold: int,
+        subscriber_identifier: Optional[str] = "msisdn",
+        tables: Optional[List[str]] = None,
```

Comment:
I think it would be good to expose the `hours` parameter here as well (and pass it down to the `TotalActivePeriodsSubscriber` queries) - sometimes we might want to use this query to find subscribers who are regularly active specifically between certain hours.